projects/plugins/jetpack/modules/videopress/class-videopress-attachment-metadata.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * Handle the VideoPress metadata properties.
+ *
+ * @package Jetpack
+ */
+
+use Automattic\Jetpack\Connection\Client;
+
+/**
+ * Class Videopress_Attachment_Metadata
+ */
+class Videopress_Attachment_Metadata {
+
+	/**
+	 * Persist the VideoPress metadata information, including rating and display_embed.
+	 *
+	 * @param string|int $post_id       The post id.
+	 * @param string     $guid          VideoPress Guid.
+	 * @param string     $post_title    The post title.
+	 * @param string     $caption       Video caption.
+	 * @param string     $post_excerpt  The post excerpt.
+	 * @param string     $rating        The rating.
+	 * @param int        $display_embed The display_embed.
+	 *
+	 * @return bool|\WP_Error
+	 */
+	public static function persist_metadata( $post_id, $guid, $post_title, $caption, $post_excerpt, $rating, $display_embed ) {
```

Comment:
The implementation of this class and method was extracted from VideoPress_Edit_Attachment::save_fields so that I could reuse the same logic for both WP-Admin and API endpoints. I did however, some changes because I had issues in WPCOM side (Which I will mention them below).